src/webhook_launcher/app/payload.py

```diff
@@ -429,3 +442,113 @@ def handle(self):
                     trigger_build(
                         mapobj, payload["user"], lsr=seenrev, tag=tag
                     )
+
+
+class BbPushV2(Payload):
+    def __init__(self, data):
+        super(BbPushV2, self).__init__(data)
+        try:
+            data['push']
+            path = data['repository']['full_name']
+        except KeyError as e:
+            raise PayloadParsingError("Not a BbPushV2 payload: %s" % e)
+
+        print("bitbucket V2 payload")
+        self.url = urlparse.urljoin("https://bitbucket.org", path) + '.git'
+
+    def handle(self):
+        branches = {}
+        tags = {}
+        user = self.data['actor']['username']
+
+        # Collect branches and tags
+        for change in self.data['push']['changes']:
+            newref = change.get('new')
+            commits = change.get('commits', [])
+            if not newref:
+                # TODO: Handle delete events
+                continue
+            name = newref['name']
+            hash = newref['target']['hash']
+            if newref['type'] == 'branch':
+                branches[name] = [hash, commits]
+            elif newref['type'] == 'tag':
+                tags[name] = [hash, set()]
+            else:
+                print("Unknown ref type '%s'" % newref['type'])
+                continue
+
+        # Find branches for tags
+        for tag, (hash, tag_branches) in tags.items():
+            # TODO: this will only find branches with head matching the tag
+            bbcall = bbAPIcall(self.data['repository']['full_name'])
+            bts = bbcall.branches_tags()
+            for branch in bts['branches']:
+                if branch['changeset'] == hash:
+                    tags[tag][1].add(branch['name'])
+
+        # Handle commits in branches
+        for branch, (revision, commits) in branches.iteritems():
+            mapobjs = WebHookMapping.objects.filter(
+                repourl=self.url, branch=branch,
+            )
+            if not len(mapobjs):
```

Comment:
Using count() does not make much difference, because the the queryset is iterated over later if there is any results. So instead of two queries for count and later to get the result, we now do just one and the Queryset object caches the result.